crates/core/src/test_env.rs

```diff
@@ -12,47 +12,30 @@ use std::marker::PhantomData;
 use std::str::FromStr;
 use std::sync::Arc;
 
+use enumset::EnumSet;
 use futures::Stream;
 
-use restate_types::cluster_controller::{ReplicationStrategy, SchedulingPlan};
+use restate_types::cluster_controller::ReplicationStrategy;
 use restate_types::config::NetworkingOptions;
-use restate_types::logs::metadata::{bootstrap_logs_metadata, ProviderKind};
-use restate_types::metadata_store::keys::{
-    BIFROST_CONFIG_KEY, NODES_CONFIG_KEY, PARTITION_TABLE_KEY, SCHEDULING_PLAN_KEY,
-};
+use restate_types::logs::metadata::ProviderKind;
 use restate_types::net::codec::{Targeted, WireDecode};
-use restate_types::net::metadata::MetadataKind;
 use restate_types::net::AdvertisedAddress;
 use restate_types::nodes_config::{LogServerConfig, NodeConfig, NodesConfiguration, Role};
-use restate_types::partition_table::PartitionTable;
 use restate_types::protobuf::node::Message;
-use restate_types::{GenerationalNodeId, Version};
+use restate_types::retries::RetryPolicy;
+use restate_types::{GenerationalNodeId, NodeId, Version};
 
-use crate::metadata_store::{MetadataStoreClient, Precondition};
+use crate::metadata_store::MetadataStoreClient;
 use crate::network::{
-    ConnectionManager, FailingConnector, Incoming, MessageHandler, MessageRouterBuilder,
-    NetworkError, Networking, ProtocolError, TransportConnect,
+    ConnectionManager, FailingConnector, Incoming, MessageHandler, NetworkError, Networking,
+    ProtocolError, TransportConnect,
 };
-use crate::{spawn_metadata_manager, MetadataBuilder, TaskId};
-use crate::{Metadata, MetadataManager, MetadataWriter};
+use crate::{CoreBuilder, CoreBuilderFinal, MetadataBuilder, StartedCore};
 use crate::{TaskCenter, TaskCenterBuilder};
 
-pub struct TestCoreEnvBuilder<T> {
-    pub tc: TaskCenter,
-    pub my_node_id: GenerationalNodeId,
-    pub metadata_manager: MetadataManager<T>,
-    pub metadata_writer: MetadataWriter,
-    pub metadata: Metadata,
-    pub networking: Networking<T>,
-    pub nodes_config: NodesConfiguration,
-    pub provider_kind: ProviderKind,
-    pub router_builder: MessageRouterBuilder,
-    pub partition_table: PartitionTable,
-    pub scheduling_plan: SchedulingPlan,
-    pub metadata_store_client: MetadataStoreClient,
-}
+pub type TestCoreEnvBuilder<T> = CoreBuilderFinal<T>;
```

Comment:
aliases keep this PR small for the time being; can rewrite these types everywhere later